supervised/automl.py

```diff
@@ -1087,11 +746,472 @@ def predict(self, X):
         elif self._ml_task == MULTICLASS_CLASSIFICATION:
             target_is_numeric = self._data_info.get("target_is_numeric", False)
             if target_is_numeric:
-                predictions["label"] = predictions["label"].astype(int)
+                predictions["label"] = predictions["label"].astype(np.int32)
             return predictions
+        # Regression
         else:
             return predictions
 
+    def _predict(self, X):
+
+        predictions = self._base_predict(X)
+        # Return predictions
+        # If classification task the result is in column 'label'
+        # If regression task the result is in column 'prediction'
+        return (
+            predictions["label"].to_numpy()
+            if self._ml_task != REGRESSION
+            else predictions["prediction"].to_numpy()
+        )
+
+    def _predict_proba(self, X):
+        # Check is task type is correct
+        if self._ml_task == REGRESSION:
+            raise AutoMLException(
+                f"Method `predict_proba()` can only be used when in classification tasks. Current task: '{self._ml_task}'."
+            )
+
+        # Make and return predictions
+        # If classification task the result is in column 'label'
+        # Need to drop `label` column.
+        return self._base_predict(X).drop(["label"], axis=1).to_numpy()
+
+    def _score(self, X, y=None):
+        # y default must be None for scikit-learn compatibility
+
+        # Check if y is None
+        if y is None:
+            raise AutoML("y must be specified.")
```

Comment:
Probably should be AutoMLException

Reply:
Yes! Thanks for noticing the typo. Gonna add a test case for when `y` is `None`

Reply:
:+1: